src/engine/Actor.ts

```diff
@@ -527,6 +527,64 @@ export class ActorImpl extends Class implements Actionable, Eventable, PointerEv
    */
   public currentDrawing: Drawable = null;
 
+  /**
+   * Draggable helper
+   */
+  private _draggable: boolean = false;
+  private _dragging: boolean = false;
+
+  public get draggable(): boolean {
+    return this._draggable;
+  }
+
+  public set draggable(isDraggable: boolean) {
+    if (isDraggable) {
+      if (isDraggable && !this._draggable) {
+        this.on('pointerdragstart', () => {
+          this._dragging = true;
+        });
+
+        this.on('pointerdragstart', () => {
+          this._dragging = false;
+        });
+
+        this.on('pointerdragmove', (pe: PointerEvent) => {
+          if (this._dragging) {
+            this.pos = pe.pointer.lastWorldPos;
+          }
+        });
+
+        this.on('pointerdragleave', (pe: PointerEvent) => {
+          if (this._dragging) {
+            this.pos = pe.pointer.lastWorldPos;
+          }
+        });
+      } else if (!isDraggable && this._draggable) {
+        this.off('pointerdragstart', () => {
+          this._dragging = true;
+        });
+
+        this.off('pointerdragstart', () => {
+          this._dragging = false;
+        });
+
+        this.off('pointerdragmove', (pe: PointerEvent) => {
```

Comment:
@MaanasArora Last request for this PR, thanks for working so hard on this. In order for `this.off(...)` to work correctly it needs to be based the same function reference that was used in the `this.on(...)`. We'll need to store those somewhere, perhaps as private methods on actor?

Something like
```typescript
private _pointerDragStartHandler = () => {
   this._dragging = true;
}
...

 public set draggable(isDraggable: boolean) {
    if (isDraggable) {
      if (isDraggable && !this._draggable) {
        this.on('pointerdragstart', this._pointerDragStartHandler );
      ...
      } else if (!isDraggable && this._draggable) {
        this.off('pointerdragstart', this._pointerDragStartHandler );
      ...
}